tor/core/user_interaction.py

```diff
@@ -74,39 +73,45 @@ def process_claim(post, config):
         logging.debug('Received `claim` on post we do not own. Ignoring.')
         return
 
-    if not coc_accepted(post, config):
-        # do not cache this page. We want to get it every time.
-        post.reply(_(
-            please_accept_coc.format(get_wiki_page('codeofconduct', config))
-        ))
-        return
+    try:
+        if not coc_accepted(post, config):
+            # do not cache this page. We want to get it every time.
+            post.reply(_(
+                please_accept_coc.format(get_wiki_page('codeofconduct', config))
+            ))
+            return
+
+        if top_parent.link_flair_text is None:
+            # There exists the very small possibility that the post was malformed
+            # and doesn't actually have flair on it. In that case, let's set
+            # something so the next part doesn't crash.
+            flair_post(top_parent, flair.unclaimed)
+
+        if flair.unclaimed in top_parent.link_flair_text:
+            # need to get that "Summoned - Unclaimed" in there too
+            post.reply(_(claim_success))
 
-    if top_parent.link_flair_text is None:
-        # There exists the very small possibility that the post was malformed
-        # and doesn't actually have flair on it. In that case, let's set
-        # something so the next part doesn't crash.
-        flair_post(top_parent, flair.unclaimed)
+            flair_post(top_parent, flair.in_progress)
+            logging.info(
+                'Claim on ID {} by {} successful'.format(
+                    top_parent.fullname, post.author
+                )
+            )
 
-    if flair.unclaimed in top_parent.link_flair_text:
-        # need to get that "Summoned - Unclaimed" in there too
-        try:
-            post.reply(_(claim_success))
-        except:
-            # Pause bubbling up the stack trace to log this error
-            logging.error('ERROR: {1} of type {0}\n{2}'.format(*sys.exc_info()), extra={'stack': True})
-            raise  # Bubble up the original exception
-
-        flair_post(top_parent, flair.in_progress)
-        logging.info(
-            'Claim on ID {} by {} successful'.format(
-                top_parent.fullname, post.author
+        # can't claim something that's already claimed
+        elif top_parent.link_flair_text == flair.in_progress:
+            post.reply(_(already_claimed))
+        elif top_parent.link_flair_text == flair.completed:
+            post.reply(_(claim_already_complete))
+
+    except praw.exceptions.APIException as e:
+        if e.error_type == 'DELETED_COMMENT':
+            logging.debug(
```

Comment:
Let's check to see if we have a top_parent object and then force-reset the flair back to unclaimed in case it crashed after awarding flair. `flair_post(top_parent, flair.unclaimed)`

Reply:
I interpreted the code being slightly different already. The diff doesn't make this too clear, but it looked like events happen in this order:

1. Check for error states (e.g., no code of conduct for user) and exit with error message if any
2. Fix error state of no status in flair. Set status to unclaimed.
3. Reply to post saying it is now claimed
4. Mark OP with flair saying it's claimed

(1) failing means it didn't modify the flair anyway, since it only checks for error states, replies with an error message, and quits out

(2) failing means it didn't correct an already failing state.

(3) failing is the main error for which we're trying to account. If it fails, (4) won't be executed anyway and flair isn't modified in a way that needs to be reverted.

(4) failing means it has a bot-replied claimed comment, but isn't actually claimed. This should never happen since the OP being deleted means either the bot (original author) or a subreddiit mod or a super mod removed it. In any event, the post is gone and, claimed or not, it cannot be completed anyway.

It keeps going for possible events, but they're also error handling by posting an error message in a reply. No modification of flair that needs to be fixed if it cuts out early.

Reply:
It looks like we modify the flair after replying to the comment already. If the reply fails, which is what throws this `DELETED_COMMENT` error, we don't even execute the flair modification code. Do we really need to reset it then?

Reply:
It appears you're right. Nevermind! :D